compiler/src/dotty/tools/repl/Rendering.scala

```diff
@@ -106,11 +106,17 @@ private[repl] class Rendering(parentClassLoader: Option[ClassLoader] = None) {
       resObj
         .getDeclaredMethods.find(_.getName == sym.name.encode.toString)
         .map(_.invoke(null))
-    val string = value.map(replStringOf(_))
+
+    val resultSym = sym.info.classSymbol
+    val valueString = if (resultSym.isValueClass && !resultSym.isPrimitiveValueClass) then
+      value.map(value => s"${sym.info.classSymbol.name}@${value.hashCode.toHexString}")
```

Comment:
This won't give the correct result if toString is overridden in the value class (as it is in case classes by default). Instead you could try using reflection to do `new ValueClass(theUnboxedValue)` prior to calling replStringOf on it.